nexus/src/db/ipv6.rs

```diff
@@ -32,16 +33,10 @@ impl From<&std::net::Ipv6Addr> for Ipv6Addr {
     }
 }
 
-impl<DB> ToSql<Inet, DB> for Ipv6Addr
-where
-    DB: Backend,
-    IpNetwork: ToSql<Inet, DB>,
-{
-    fn to_sql<W: std::io::Write>(
-        &self,
-        out: &mut Output<W, DB>,
-    ) -> serialize::Result {
-        IpNetwork::V6(Ipv6Network::from(self.0)).to_sql(out)
+impl ToSql<Inet, Pg> for Ipv6Addr {
+    fn to_sql<'a>(&'a self, out: &mut Output<'a, '_, Pg>) -> serialize::Result {
+        let net = IpNetwork::V6(Ipv6Network::from(self.0));
+        <IpNetwork as ToSql<Inet, Pg>>::to_sql(&net, &mut out.reborrow())
```

Comment:
Does this `as` does the same work as the `where` in the previous version, or is there any subtle difference?

Reply:
In the old version, the where clause specified "`IpNetwork` must implement the `ToSql<Inet, Db>` trait for this to be compiled". As you alluded to, this was necessary because the code could be compiled for any `DB` backend.

In the new version, `ToSql<Inet, Pg>` is not a generic trait, and `IpNetwork` is not a generic structure - so we have no need to specify this bound.

Instead, the `as` expression here is used to disambiguate. `ToSql<SomeOtherStructure, Pg>` could also be implemented for `IpNetwork`, so calling `IpNetwork.to_sql(...)` is vague - there could be multiple serialization output types. By using this "fully qualified" syntax, only one option is picked.